src/add-text-track-data.js

```diff
@@ -66,19 +66,38 @@ const addTextTrackData = function(sourceHandler, captionArray, metadataArray) {
   if (metadataArray) {
     metadataArray.forEach(function(metadata) {
       let time = metadata.cueTime + this.timestampOffset;
+      let videoDuration = Number.isFinite(this.mediaSource_.duration) ?
```

Comment:
I wonder if Babel will polyfill `Number.isFinite` in older browsers? This could also be written as `let videoDuration = Math.min(this.mediaSource_.duration, Number.MAX_VALUE);`.